src/Microsoft.AspNet.Mvc.Core/Rendering/Html/DefaultDisplayTemplates.cs

```diff
@@ -91,7 +91,7 @@ internal static List<SelectListItem> TriStateValues(bool? value)
 
         public static string CollectionTemplate(IHtmlHelper htmlHelper)
         {
-            var model = htmlHelper.ViewData.ModelMetadata.Model;
+            var model = htmlHelper.ViewData.Model;
```

Comment:
This would have been possible due to some quirks of of VDD works (`ModelMetadata` is shared and mutable = sadness). 

We should nuke this lack of clarity from orbit with a bomb that fires jackhammers. Right now all the tests are passing. If we find bugs here we'll fix them. Having two similar but non-equivalent ways to get at something is just begging for pain.